### ISSUE TITLE
azure's blob name must be 23 character long

### ISSUE THREAD

**Issue by nicocesar** [mitchellh/packer#3551]
I ended myself in a situation where the "capuring image" was unable to finish because the name f48sn-workbench-1463594889 was too long.  response is very confusing unless PACKER_LOG is enabled. 

please review the sanity check of  capture_name_prefix accordingly so this error can be spotted early on 

```
2016/05/18 18:25:09 packer: 2016/05/18 18:25:09 Azure request method="POST" request="https://management.azure.com/subscriptions/594bef42-33f3-42df-b056-e7399d6ae7a0/resourceGroups/packer-Resource-Group-16z9420dw1/providers/Microsoft.Compute/virtualMachines/pkrvm16z9420dw1/generalize?api-version=2015-06-15" body=""
2016/05/18 18:25:10 packer: 2016/05/18 18:25:10 Azure response status="200 OK" method="POST" request="https://management.azure.com/subscriptions/594bef42-33f3-42df-b056-e7399d6ae7a0/resourceGroups/packer-Resource-Group-16z9420dw1/providers/Microsoft.Compute/virtualMachines/pkrvm16z9420dw1/generalize?api-version=2015-06-15" x-ms-request-id="6c00d668-37b0-4a41-a935-abda3fdfe99b" body=""
2016/05/18 18:25:10 packer: 2016/05/18 18:25:10 Azure response status="200 OK" method="POST" request="https://management.azure.com/subscriptions/594bef42-33f3-42df-b056-e7399d6ae7a0/resourceGroups/packer-Resource-Group-16z9420dw1/providers/Microsoft.Compute/virtualMachines/pkrvm16z9420dw1/generalize?api-version=2015-06-15" x-ms-request-id="6c00d668-37b0-4a41-a935-abda3fdfe99b" body=""
2016/05/18 18:25:10 packer: 2016/05/18 18:25:10 Azure request method="POST" request="https://management.azure.com/subscriptions/594bef42-33f3-42df-b056-e7399d6ae7a0/resourceGroups/packer-Resource-Group-16z9420dw1/providers/Microsoft.Compute/virtualMachines/pkrvm16z9420dw1/capture?api-version=2015-06-15" body="{\"vhdPrefix\":\"f48sn-workbench-1463594889\",\"destinationContainerName\":\"images\",\"overwriteVhds\":false}"
2016/05/18 18:25:10 packer: 2016/05/18 18:25:10 Azure response status="400 Bad Request" method="POST" request="https://management.azure.com/subscriptions/594bef42-33f3-42df-b056-e7399d6ae7a0/resourceGroups/packer-Resource-Group-16z9420dw1/providers/Microsoft.Compute/virtualMachines/pkrvm16z9420dw1/capture?api-version=2015-06-15" x-ms-request-id="66794f59-7dfb-4224-bdaa-4c6a9b7ebeb1" body="{\r\n  \"error\": {\r\n    \"code\": \"InvalidParameter\",\r\n    \"target\": \"vhdPrefix\",\r\n    \"message\": \"'f48sn-workbench-1463594889' is not a valid captured VHD blob name prefix. A valid prefix matches regex '^[^_\\\\W][\\\\w-._]{0,23}(?<![-.])$'.\"\r\n  }\r\n}"
2016/05/18 18:25:10 ui error: ==> azure-arm: azure#updatePollingState: Azure Polling Error - Unable to obtain polling URI for POST : StatusCode=0
==> azure-arm: azure#updatePollingState: Azure Polling Error - Unable to obtain polling URI for POST : StatusCode=0
```

- Packer Version: 0.10.1
- Host platform: GNU/Linux 3.2.0-4-amd64 Debian 3.2.65-1+deb7u1 x86_64

**Comment by nicocesar** [comment]
please add label:builder/azure

**Comment by boumenot** [comment]
Thank you for reporting, and including **good** details in the initial report.  This is a duplicate of #3535.  There's a [PR](https://github.com/mitchellh/packer/pull/3537) pending to fix the issue.

If you're able, would you please see if this version rejects the name before starting the build.
